js/src/wrappers/openai-agents.ts

```diff
@@ -0,0 +1,343 @@
+  type: string;
+  traceId: string;
+  name: string;
+  groupId?: string;
+  metadata?: Record<string, any>;
+}
+
+interface AgentsSpan {
+  type: string;
+  spanId?: string;
+  traceId?: string;
+  name?: string;
+  spanData?: {
+    type?: string;
+    name?: string;
+    tools?: any[];
+    handoffs?: any[];
+    outputType?: string;
+    input?: any;
+    output?: any;
+    response?: {
+      output?: any;
+      metadata?: Record<string, any>;
+      usage?: {
+        totalTokens?: number;
+        inputTokens?: number;
+        outputTokens?: number;
+        promptTokens?: number;
+        completionTokens?: number;
+      };
+    };
+    usage?: Record<string, any>;
+    model?: string;
+    modelConfig?: Record<string, any>;
+    fromAgent?: string;
+    toAgent?: string;
+    triggered?: boolean;
+    data?: Record<string, any>;
+  };
+  startedAt?: string;
+  endedAt?: string;
+  error?: string;
+}
+
+function spanTypeFromAgents(span: AgentsSpan): SpanTypeAttribute {
+  const spanType = span.spanData?.type;
+
+  if (spanType === "agent" || spanType === "handoff" || spanType === "custom") {
+    return SpanTypeAttribute.TASK;
+  } else if (spanType === "function" || spanType === "guardrail") {
+    return SpanTypeAttribute.TOOL;
+  } else if (spanType === "generation" || spanType === "response") {
+    return SpanTypeAttribute.LLM;
+  } else {
+    return SpanTypeAttribute.TASK;
+  }
+}
+
+function spanNameFromAgents(span: AgentsSpan): string {
+  const spanData = span.spanData;
+
+  if (spanData?.name) {
+    return spanData.name;
+  }
+
+  switch (spanData?.type) {
+    case "generation":
+      return "Generation";
+    case "response":
+      return "Response";
+    case "handoff":
+      return "Handoff";
+    case "agent":
+    case "function":
+    case "guardrail":
+    case "custom":
+      return spanData.name || "Unknown";
+    default:
+      return "Unknown";
+  }
+}
+
+function timestampElapsed(end?: string, start?: string): number | undefined {
+  if (!start || !end) return undefined;
+  const startTime = new Date(start).getTime();
+  const endTime = new Date(end).getTime();
+  return (endTime - startTime) / 1000;
+}
+
+/**
+ * `BraintrustTracingProcessor` is a tracing processor that logs traces from the OpenAI Agents SDK to Braintrust.
+ *
+ * Args:
+ *   logger: A `Span`, `Experiment`, or `Logger` to use for logging.
+ *     If `undefined`, the current span, experiment, or logger will be selected exactly as in `startSpan`.
+ */
+export class BraintrustTracingProcessor {
+  private logger?: Span | Experiment | Logger<any>;
+  private spans: Map<string, Span> = new Map();
+
+  constructor(logger?: Span | Experiment | Logger<any>) {
+    this.logger = logger;
+  }
+
+  onTraceStart(trace: AgentsTrace): Promise<void> {
+    // Check if we already have a span for this trace to avoid duplicates
+    if (this.spans.has(trace.traceId)) {
+      return Promise.resolve();
+    }
+
+    if (this.logger) {
```

Comment:
this stanza has a lot of duplicated code. Why not something like `

```
logger = this.logger || globalLogger
logger.startSpan(...)
```

I think we can find a way to do it.